client/rpc/src/eth.rs

```diff
@@ -160,24 +155,53 @@ fn transaction_build(
 			Keccak256::digest(&rlp::encode(&transaction)).as_slice()
 		),
 		nonce: transaction.nonce,
-		block_hash: Some(H256::from_slice(
-			Keccak256::digest(&rlp::encode(&block.header)).as_slice()
-		)),
-		block_number: Some(block.header.number),
-		transaction_index: Some(U256::from(
-			UniqueSaturatedInto::<u32>::unique_saturated_into(
-				status.transaction_index
-			)
-		)),
-		from: status.from,
-		to: status.to,
+		block_hash: match &block {
+			Some(block) => Some(H256::from_slice(
```

Comment:
Maybe `Option`'s `map` function would be useful here https://doc.rust-lang.org/std/option/enum.Option.html#method.map

Reply:
Updated, although I'd say from the readibility POV is a slight improvement.

For example:

```
match &block {
	Some(block) => Some(H256::from_slice(
		Keccak256::digest(&rlp::encode(&block.header)).as_slice()
	)),
	_ => None,
}
```
to

```
block.as_ref().map_or(None, |block| {
	Some(H256::from_slice(
		Keccak256::digest(&rlp::encode(&block.header)).as_slice()
	))
})
```